### ISSUE TITLE
Smalltalk code detected as HTML

### ISSUE THREAD

**Issue by jecisc** [github/linguist#3244]
Most of my repositories are in Smalltalk but it happen a lot that files are detected as HTML.

For example on this project: https://github.com/LeCoinEnBande/OOnoz
Currently it is considered as 85.8% Smalltalk, 14% HTML and 0.2% CSS but in reality it is 99.8% Smalltalk.

Examples of mismatched code:
- https://raw.githubusercontent.com/LeCoinEnBande/OOnoz/02527b40d4c1a2f09c6b40607996ab0197075084/src/OOnoz-WebApplication.package/OOFooter.class/instance/renderLeftSectionContentOn..st
- https://github.com/LeCoinEnBande/OOnoz/raw/0e66dc0d80bac5a31f066935463d16979959ffb5/src/BaselineOfOOnoz.package/monticello.meta/categories.st
- https://github.com/LeCoinEnBande/OOnoz/raw/27c361d6b7fe783d36975cd956afd201ecbb4050/src/OOnoz-Server.package/OOConfiguration.class/class/stonName.st
- https://github.com/LeCoinEnBande/OOnoz/raw/02527b40d4c1a2f09c6b40607996ab0197075084/src/OOnoz-WebApplication.package/OOFooter.class/instance/renderContentOn..st
- https://github.com/LeCoinEnBande/OOnoz/raw/02527b40d4c1a2f09c6b40607996ab0197075084/src/OOnoz-WebApplication.package/OOHeader.class/instance/renderContentOn..st
- https://raw.githubusercontent.com/LeCoinEnBande/OOnoz/02527b40d4c1a2f09c6b40607996ab0197075084/src/OOnoz-WebApplication.package/OOLibrary.class/instance/dialogpolyfillminCss.st
- https://github.com/LeCoinEnBande/OOnoz/raw/023cb6fdd5cfda04f1eb026edfe922d9076e649e/src/OOnoz-Development.package/OODevHelper.class/class/installGitFileTree.st

**Comment by jecisc** [comment]
I saw that in the languages.yml, HTML declare ".st" as a possible extension. I cannot find why this should be an HTML extension. Is that an error? Or is there a reason?

**Comment by pchaigno** [comment]
I added the `.st` extension to HTML in #1117.
Do you see a lot of misclassified Smalltalk files on github.com in general? If the issue is limited to your repository, the easiest solution would be to use [Linguist overrides](https://github.com/github/linguist#overrides).

**Comment by jecisc** [comment]
@pchaigno This is a general issue.

In all the projects I checked I found some misclassified files. 

DuneST/MaterialDesignLite : 4.4% misclassified
jecisc/SeasideComponents : 8.3% misclassified
jecisc/WebPokedex : 1.5% misclassified
npasserini/iceberg : 3.4% misclassified
LeCoinEnBande/OOnoz : 14% misclassified
DuneSt/ChartJs : 57.3% misclassified 
DuneSt/PrismCodeDisplayer : 96.6% misclassified

If the .st extension for html is justified, maybe only adding the samples I added will fix the problem. *fingers crossed*
